### ISSUE TITLE
Provide service files for atuin server?

### ISSUE THREAD

**Issue by LecrisUT** [SriRamanujam/atuin-rpmspec#2]
Still trying to quick-start atuin, but the first point I see is that it defaults to connecting to an external server. It seems that it also support running a self-contained server. It would be good to provide a unit-file to launch the server locally:

https://docs.atuin.sh/self-hosting/server-setup/

**Comment by SriRamanujam** [comment]
That's a good idea, thank you for the suggestion!

**Comment by LecrisUT** [comment]
I have been giving this a few tries, and I think we need to work more on the defaults and required packages
- With plain bash, it does not seem to save the recent history as you go. The documentation seems to suggest using either `ble.sh` or `bash-preexec`. `atuin` seems to prefer the former. I can confirm that using either of them fixes this issue. Both of them seem to be simple shell scripts that can easily be packaged, but I would vote for `bash-preexec` because it has less side-effects
- Running a server locally, seems to be orthogonal with the topic above, and entirely optional. Seems to make sense more if the history is to be shared across devices
- Consider writing a file to `/etc/profile.d/atuin.sh` with
  ```bash
  eval "$(atuin init bash)"
  ```
  For security purposes, I would sugest to copy the content of `atuin init bash` instead of using eval directly there. I did confirm that it works as expected with `bash-preexec.sh` and `atuin.sh` in `/etc/profile.d`
- Maybe some tests should be written because this has the potential to cripple a terminal. But this I think I will consult with the folks at `tmt` and `testing-farm` on how to approach after the package is uploaded to Fedora.

**Comment by SriRamanujam** [comment]
> Running a server locally, seems to be orthogonal with the topic above, and entirely optional. Seems to make sense more if the history is to be shared across devices

Yeah, I use a self-hosted server myself. It would be good to include the service file anyway, I think, as it's the same binary for both server and client. The other alternative would be a tiny `atuin-server` package that just shipped the unit file and required the main `atuin` package as a dependency, which seems like overkill.

**Comment by LecrisUT** [comment]
> The other alternative would be a tiny atuin-server package that just shipped the unit file and required the main atuin package as a dependency, which seems like overkill.

There are other things that can be managed there:
- `systemctl enable` scriplets
- pointing atuin to the self-hosted endpoint (this one would be tricky)

I didn't research reference packages to see how they do it. I think a `go`, `rust`, `python/flask` could have some reference, but I can't think of a project that can be run self-contained like this.

**Comment by SriRamanujam** [comment]
> pointing atuin to the self-hosted endpoint (this one would be tricky)

I don't think we should configure it like this, tbh. Those who are interested in self-hosting will almost certainly be running the server and client on separate computers. For that, a simple service file will be sufficient, and then `atuin login` walks you through the rest. 

For now, I think we start with a service file (easy peasy) and _maybe_ a preset? I seem to recall that Fedora wants services to ship disabled, so maybe we don't need a preset at all since that's the upstream default.

> I can't think of a project that can be run self-contained like this.

Yeah, I can't either. Nothing that's packaged, at least.
